csrc/multidevice/communicator.cpp

```diff
@@ -64,12 +66,35 @@ int parseEnv(
   if (!env) {
     env = std::getenv("WORLD_LOCAL_SIZE");
     if (!env) {
-      return 1;
+      return false;
     }
   }
   local_size = std::atoi(env);
 
-  return 0;
+  // retrieves master address
+  env = std::getenv("MASTER_ADDR");
```

Comment:
Does MPI always provide this, even on a node not connected to any network?

Reply:
No, MPI never provides it, the user must provide it

Reply:
At some point, we would want to use python frontend anyway
https://github.com/NVIDIA/Fuser/issues/639

Reply:
MPI doesn't provide `MASTER_ADDR`, but you can query world size and world rank using `MPI_Comm_size` and `MPI_Comm_rank`. Why are these variables expected to be defined with environment variables?

Reply:
> MPI doesn't provide `MASTER_ADDR`

Correct. However a master address is needed to create a pytorch Store (necessary for Process Groups).
If you use torchrun though, `MASTER_ADDR` will be automatically set. 

> you can query world size and world rank using MPI_Comm_size and MPI_Comm_rank

We don't directly use OpenMPI library in nvFuser, we do everything through pytorch's process group. So the functions MPI_Comm_size or MPI_Comm_rank are not available.
The environment variables from which we retrieve the rank and size are always assumed to be set in MPI processes (also when using OpenMPI library), that is why we use it.

Reply:
Thank you for the explanations!